### DISCUSSION TITLE
Mirror fin in x-z-plane

### DISCUSSION THREAD

**Discussion by zmmn** [DLR-SL/CPACS#785] [Q&A]
Is it possible to create a (symmetrical) fin with only half of the airfoil points and to mirror the rest in the x-z-plane?
My attempts were not successful: TiGL viewer crashes.

However TiGL viewer depicts the fin correctly without crashes, if all airfoil points are in the list. - But then the fin is not mirrored. 

In the CPACS documentation of v3.3 it is documented, that the representation of symmetric airfoils is possible.
My point lists have roughly the following format:

X: 1; ...0.5;   ...;0;
Y: 0; ...0;     ...;0;
Z: 0; ...-0.5; ...;0;

**Comment by joergbrech** [comment]
Hello @zmmn, 

short answer: No, symmetrical wings with only half of the airfoil points are currently not supported by TiGL. This feature currently only exists *(in TiGL)* for fuselages and is not widely used as far as I know.

Do you have a specific use case? I believe most people that need a half-model usually create a full airplane with TiGL and then use Boolean operations to cut away the symmetric part. TiGL has some convenience functionality to subtract the fused (full) airplane from a far field to get a negative half-model for aerodynamic simulations.

If you add all the points to your fin profile, the symmetry attribute of the airfoil is simply ignored *(though I suppose it would make sense to log a warning in this situation)*. However, TiGL should not crash, but rather give you a clear error message. If you could provide a minimal example and open an issue here: https://github.com/DLR-SC/tigl/issues it would be much appreciated.
